controllers/ingestion/http/http.go

```diff
@@ -0,0 +1,79 @@
+package http
+
+import (
+	"bytes"
+	"io/ioutil"
+	"net/http"
+)
+
+// DruidHTTP interface
+type DruidHTTP interface {
+	Do() (*Response, error)
+}
+
+// HTTP client
+type Client struct {
```

Comment:
This looks more like a `Request` object rather than a `Client` object.

Reply:
I suggest removing the method, URL, and body from the client. Also, change the object's name to `DruidClient` Since this is an HTTP client specific for Druid (the `Auth` object is the difference).